setup.py

```diff
@@ -0,0 +1,21 @@
+from setuptools import setup
+
+requirements = [
+    "fsspec",
+    "numpy",
+    "numcodecs",
```

Comment:
Do we know what versions are needed Alistair?

Reply:
Not sure what minimum versions would be. Do we need to pin a minimum version or can we leave this open?

Reply:
I think they can be left open. I was in two minds about whether this should go in at all as *lack* of build files might be a useful tool to dissuade people from using it for the time being. But certainly, there should be no guarantees for support and it shouldn't be included with other projects, so loose dependencies (just enough that a clean, modern python environment could `pip install` it from github or the directory) should be fine.

Reply:
I think they can be left open. Just wanted to make sure we didn't get too old of a version by accident. Guess we could just pick latest whenever this gets tagged.